File: palindromo.js
/*Determinar si una cadena de caracteres"""
es un palíndromo
Un palíndromo es una palabra o enunciado 
que es escrito de la misma forma de adelante
 y al-revés

Ejemplos:
 -Ejecuto esta función:
 palindromo("otto") // Debe devolver true.
 palindromo("victor") // Debe devolver false.

 Crea una función que regresa verdadero si la 
cadena de texto es un palíndromo.  De otra 
forma, regresa falso.
Nota. La salida de la función debe ser en consola
Se recomienda certificar el funcionamiento en 
replit.com*/

function Palindromo(str) {
    const Reversa = str.split("").reverse().join(""); //se usa split para separar la palabra, reverse para recorrer la palabra
    //y join para unir de nuevo la palabra
    return Reversa === str ? "la palabra es un palindromo" : "la palabra no es un palindromo" //este return devuelve si es un palindromo o no lo es

    
}

console.log(Palindromo("otto"));
console.log(Palindromo("victor"))

File: palindromo.py

def Palindromo(str):
    reversa= "".join(reversed(str))
    return "Es un palindromo" if str==reversa else "No es palindromo"

def esPalindromo(str):
    longuitud=len(str)
    for i in range (longuitud//2):
        if str[i]!=str[longuitud-1-i]:
            return False


    return True


def Palin(str):
    if str == str[::-1]:
        return True
    else:
        return False



str = "otto"
#str="hola"
#str="Victor"
#str="ala"

if Palin(str):
    print(f"El string",{str},"es palíndrome")
else:
    print (f"El string ",{str}, "no es palíndrome.")











print(Palindromo("otto"))
print(Palindromo("hola"))
print(Palindromo("Victor"))
print(Palindromo("ala"))


print(esPalindromo("otto"))
print(esPalindromo("hola"))
print(esPalindromo("Victor"))
print(esPalindromo("ala"))



